android/fakedapp/src/main/java/com/solana/mobilewalletadapter/fakedapp/MainViewModel.kt

```diff
@@ -519,6 +538,7 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
             try {
                 sender.startActivityForResult(associationIntent) {
                     viewModelScope.launch {
+                        releaseLock()
```

Comment:
We'll need to rework the callback structure a bit to move the invocation of this closure into ViewModel, to avoid any lifecycle issues stemming from transient Activity re-creation.